WalletWasabi/WabiSabi/Backend/Rounds/Arena.cs

```diff
@@ -663,10 +662,9 @@ private Script GetCoordinatorScriptPreventReuse(Round round)
 	private void CoinVerifier_CoinBlacklisted(object? _, Coin coin)
 	{
 		// For logging reason Prison needs the roundId.
-		var roundState = RoundStates.FirstOrDefault(rs => rs.CoinjoinState.Inputs.Any(input => input.Outpoint == coin.Outpoint));
+		var round = Rounds.ToList().FirstOrDefault(round => round.Alices.Select(alice => alice.Coin.Outpoint).Contains(coin.Outpoint));
```

Comment:
I don't know about this. First of all, we need to "snapshot" Alices as well. Officially creating a snapshot should not happen here, but whenever at the place where we touch Alices, because that is inside the lock. 

My concern here is that this can cause issues not just only here but in places where we are using the list - leading to functional issues. I am not sure how it can cause issues - but until there is no proof I don't want to sacrifice CJ stability in fixing log messages. 

So here are the options as far as I can see.

- We do not touch this.
- You create a snapshot every time you touch Alices into somewhere and use that here. 
- You research and somehow prove that accessing List from multiple threads cannot cause problems.